app/services/data_api/tad_application_export.rb

```diff
@@ -47,19 +47,19 @@ def as_json
         degree_classification_hesa_code: degree&.grade_hesa_code,
 
         # Provider
-        provider_code: application_choice.provider.code,
-        provider_id: application_choice.provider.id,
-        provider_name: application_choice.provider.name,
+        provider_code: application_choice.current_provider.code,
+        provider_id: application_choice.current_provider.id,
+        provider_name: application_choice.current_provider.name,
         accrediting_provider_code: accrediting_provider.code,
         accrediting_provider_id: accrediting_provider.id,
         accrediting_provider_name: accrediting_provider.name,
 
         course_level: course.level,
-        program_type: application_choice.course.program_type,
-        programme_outcome: application_choice.course.description,
-        course_name: application_choice.course.name,
-        course_code: application_choice.course.code,
-        nctl_subject: concatenate(application_choice.course.subject_codes),
+        program_type: application_choice.current_course.program_type,
+        programme_outcome: application_choice.current_course.description,
+        course_name: application_choice.current_course.name,
+        course_code: application_choice.current_course.code,
+        nctl_subject: concatenate(application_choice.current_course.subjects.map(&:code)),
```

Comment:
could still use `subject_codes` here?

Reply:
No — for some reason that causes n+1. Will remove subject_codes in followup if it's unused elsewhere